rtran-maven/src/main/scala/com/ebay/rtran/maven/util/MavenUtil.scala

```diff
@@ -96,6 +96,12 @@ object MavenUtil {
     val dependencyFilter = new ExclusionsDependencyFilter(
       dependency.getExclusions.map(e => s"${e.getGroupId}:${e.getArtifactId}")
     )
+
+    def recoverFromResolutionException: PartialFunction[Throwable, util.List[Artifact]] = {
```

Comment:
Actually it does make sense. You PartialFunction is defined as [Throwable, util.List[Artifact]]. However, the case you matches is only for `DependencyResolutionException` which is only a subclass of `Throwable`. If other exceptions occurs, you will get a match exception. You can do
```scala

def recoverFromResolutionException: PartialFunction[Throwable, util.List[Artifact]] = {
      case e: DependencyResolutionException =>
        e.getResult.getArtifactResults.flatMap(r => Option(r.getArtifact))
      case other: Throwable => throw other;
    }
```